src/legend_data_monitor/analysis_data.py

```diff
@@ -126,10 +126,13 @@ def __init__(self, sub_data: pd.DataFrame, **kwargs):
             "det_type",
             "status",
         ]
-        # pulser flag is present only if subsystem.flag_pulser_events() was called
-        # needed to subselect phy/pulser events
-        if "flag_pulser" in sub_data:
-            params_to_get.append("flag_pulser")
+        for col in sub_data.columns:
+            # pulser flag is present only if subsystem.flag_pulser_events() was called -> needed to subselect phy/pulser events
+            if "flag_pulser" in col:
```

Comment:
small thing, but better either 1) `if col == "flag_pulser": append("flag_pulser")` or 2) `if "flag_pulser" in col:  append(col)`. This way to errors introduced in the future with any potential pulser flag changes